src/query/graphite/storage/m3_wrapper.go

```diff
@@ -94,6 +96,32 @@ func NewM3WrappedStorage(
 func TranslateQueryToMatchersWithTerminator(
 	query string,
 ) (models.Matchers, error) {
+	if strings.Contains(query, "**") {
+		// Need to regexp on the entire ID since ** matches over different
+		// graphite path dimensions.
+		globOpts := graphite.GlobOptions{
+			AllowMatchAll: true,
+		}
+		idRegexp, _, err := graphite.ExtendedGlobToRegexPattern(query, globOpts)
+		if err != nil {
+			return nil, err
+		}
+		// Also make sure it's a graphite metric by making sure it has a
+		// __g0__ tag:
+		hasFirstPathMatcher, err := convertMetricPartToMatcher(0, wildcard)
```

Comment:
this check seems can happen first, before the glob handling?

Reply:
Sure thing.